src/libraries/System.Security.Cryptography.Cose/tests/CoseHeaderMapTests.cs

```diff
@@ -0,0 +1,248 @@
+            // [ +label ] (non-empty array)
+            Assert.Throws<NotSupportedException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.Crit), encodedNullValue));
+            // tstr / uint
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.ContentType), encodedNullValue));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.Kid), encodedNullValue));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.IV), encodedNullValue));
+            // bstr
+            Assert.Throws<InvalidOperationException>(() => map.SetEncodedValue(new CoseHeaderLabel(KnownHeaders.PartialIV), encodedNullValue));
+        }
+
+        [Fact]
+        public void Enumerate()
+        {
+            var alg = new CoseHeaderLabel(KnownHeaders.Alg);
+            var crit = new CoseHeaderLabel(KnownHeaders.Crit);
+            var contentType = new CoseHeaderLabel(KnownHeaders.ContentType);
+            var kid = new CoseHeaderLabel(KnownHeaders.Kid);
+            var iV = new CoseHeaderLabel(KnownHeaders.IV);
+            var partialIV = new CoseHeaderLabel(KnownHeaders.PartialIV);
+
+            var map = new CoseHeaderMap();
+            map.SetValue(alg, (int)ECDsaAlgorithm.ES256);
+            //map.SetEncodedValue(crit, GetDummyCritHeaderValue());
+            map.SetValue(contentType, ContentTypeDummyValue);
+            map.SetValue(kid, s_SampleContent);
+            map.SetValue(iV, ReadOnlySpan<byte>.Empty);
+            map.SetValue(partialIV, ReadOnlySpan<byte>.Empty);
+
+            var writer = new CborWriter();
+            int currentHeader = KnownHeaders.Alg;
+            foreach ((CoseHeaderLabel Label, ReadOnlyMemory<byte> EncodedValue) in map)
+            {
+                if (currentHeader == KnownHeaders.Crit) // TODO remove when Crit gets supported
+                {
+                    currentHeader++;
+                }
+
+                Assert.Equal(new CoseHeaderLabel(currentHeader), Label);
+                ReadOnlyMemory<byte> ExpectedValue = currentHeader switch
+                {
+                    KnownHeaders.Alg => EncodeInt32((int)ECDsaAlgorithm.ES256, writer),
+                    KnownHeaders.Crit => GetDummyCritHeaderValue(),
+                    KnownHeaders.ContentType => EncodeString(ContentTypeDummyValue, writer),
+                    KnownHeaders.Kid => EncodeBytes(s_SampleContent, writer),
+                    KnownHeaders.IV or KnownHeaders.PartialIV => EncodeBytes(ReadOnlySpan<byte>.Empty, writer),
+                    _ => throw new InvalidOperationException()
+                };
+                Assert.True(ExpectedValue.Span.SequenceEqual(EncodedValue.Span));
```

Comment:
```suggestion
                AssertExtensions.SequenceEqual(expectedValue.Span, encodedValue.Span);
```

(We might have a memory/memory overload, so you might not need the `.Span`s)